zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/AbstractFormCreatedApplierTest.java

```diff
@@ -123,6 +123,7 @@ Consumer<PersistedForm> isEqualToFormRecord(final FormRecord record) {
       assertThat(bufferAsArray(persistedForm.getChecksum())).isEqualTo(record.getChecksum());
       assertThat(bufferAsArray(persistedForm.getResource())).isEqualTo(record.getResource());
       assertThat(persistedForm.getDeploymentKey()).isEqualTo(record.getDeploymentKey());
+      assertThat(persistedForm.getVersionTag()).isEqualTo(record.getVersionTag());
```

Comment:
❌ I think this change will require a new version of the applier, as (form deletion command) records from older Zeebe versions might be processed. 

@korthout just pinging you here to confirm this.

Reply:
👍 @koevskinikola This specific change is fine.

This change only ensures that two FormRecord objects are equivalent. Older `Form:CREATED` records will not have the version tag property set, and so it will be considered to contain the default `""` empty string. These should be taken into account when comparing them.

https://github.com/camunda/camunda/blob/078a73566bb5fbdffe20414090b3683f02ce9279/zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/deployment/FormRecord.java#L34

Note that the version tag is only stored in the new column family when the new v2 `From:CREATED` is appended. So I think that should be fine. 

❌ The only thing I noticed, is that we are still appending v1 of the `Form:CREATED` event in `FormResourceTransformer` and `DeploymentCreateProcessor`.

Reply:
> The only thing I noticed, is that we are still appending v1 of the Form:CREATED event in `FormResourceTransformer` and `DeploymentCreateProcessor`.

Hi @korthout,
Actually, I think version 2 is used, because it's the latest available one. There's also a test that checks the record version:

https://github.com/camunda/camunda/blob/008d8b9106088d19d4895936b6846254d019b1c9/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/FormDeploymentTest.java#L107

Reply:
Ah, shoot! Of course! I forgot about this 🙇 Thanks